### ISSUE TITLE
List of failing splitting apps

### ISSUE THREAD

**Issue by alemoi** [Starcounter/Starcounter#3154]
After splitting small apps (apps_test.bat is now liquidated) the following tests are failing: 
* 3LevelSchemaChange
```
Assertion used in a test application failed.

at Starcounter.ScAssertion.Assert(Boolean checkTrue) in C:\TeamCity\TeamCity8\buildAgent\work\sc-develop-daily-604\Level1\src\Starcounter.Logging\ScAssertion.cs:line 20
[23:13:45]   at Program.AssertMetadata() in c:\TeamCity\TeamCity8\buildAgent\work\sc-develop-daily-604\Level1\src\Samples\TestApps\3LevelSchemaChange\3LevelSchemaChange.cs:line 102
[23:13:45]   at Program.Main() in c:\TeamCity\TeamCity8\buildAgent\work\sc-develop-daily-604\Level1\src\Samples\TestApps\3LevelSchemaChange\3LevelSchemaChange.cs:line 24
```

* SelectOnNull2362
```
Assertion used in a test application failed. At least one person with LastName = null should be found.

at Starcounter.ScAssertion.Assert(Boolean checkTrue, String message) in C:\TeamCity\TeamCity8\buildAgent\work\sc-develop-daily-604\Level1\src\Starcounter.Logging\ScAssertion.cs:line 12
[23:16:44]   at Program.Main() in c:\TeamCity\TeamCity8\buildAgent\work\sc-develop-daily-604\Level1\src\Samples\TestApps\SelectOnNull2362\SelectOnNull2362.cs:line 16
```

Failing build example (branch develop-alexey):

https://scbuildserver/viewLog.html?buildId=32386&buildTypeId=Starcounter_DevelopDaily&tab=testsInfo

Here is the original issue:

https://github.com/Starcounter/CompanyTrack/issues/156#issuecomment-161771099

**Comment by k-rus** [comment]
> 3LevelSchemaChange

This test works.


> SelectOnNull2362

Seems to be a bug. For the following data:

```c#
		Db.Transact(delegate {
			p = new Person { FirstName = "Fname" };
			a = new Account { AccountId = 1 };
			d = new DailyAccount { AccountId = 2, Amount = 100.0m };
		});
```
the following query finds no records, while it is expected to find the person:
```c#
Db.SQL<Person>("select p from Person p where LastName = ?", null).First
```

**Comment by k-rus** [comment]
> SelectOnNull2362
> Seems to be a bug.

The query optimiser produces correct query plan to utilize `FullTableScan` with equality and value NULL, however created kernel iterator returns invalid record with Oid = 0, i.e., not found. Either Level1 gives incorrect input to the kernel or the kernel fails.

To reproduce run `star SelectOnNull2362.cs` or `SelectOnNull2362_test.bat` from `Level1\src\Samples\TestApps\SelectOnNull2362\`

cc @alemoi, @eriohl

**Comment by eriohl** [comment]
I can look at SelectOnNull2362 once #3278 is in place.

**Comment by eriohl** [comment]
Misunderstanding perhaps. It fails in develop?

**Comment by alemoi** [comment]
There was a separate issue regarding SelectOnNull2362 that I investigated after @k-rus report. Seems that problem was in kernel filtering null string values.

**Comment by alemoi** [comment]
#2362

**Comment by eriohl** [comment]
OK. I got the impression that it might be related to changes in QP in pnext. Ignore my previous comments please.
